src/Tablebot/Plugins/Quote.hs

```diff
@@ -270,29 +309,158 @@ deleteQ qId m =
-      )
+  return
+    ( (messageDetailsBasic t)
+        { messageDetailsEmbeds =
+            Just
+              [ addColour DiscordColorBlue $
+                  addTimestamp dtm $
+                    addFooter (pack $ "Quote #" ++ show qId) $
+                      simpleEmbed (txt <> "\n - " <> author <> maybeAddFooter link)
+              ]
+        }
+    )
   where
     getLink :: Maybe GuildId -> Maybe Text
     getLink = fmap (\x -> getMessageLink x (fromIntegral cnlId) (fromIntegral msgId))
     maybeAddFooter :: Maybe Text -> Text
     maybeAddFooter (Just l) = "\n[source](" <> l <> ") - added by " <> submitter
     maybeAddFooter Nothing = ""
 
+quoteApplicationCommand :: CreateApplicationCommand
+quoteApplicationCommand = CreateApplicationCommandChatInput "quote" "store and retrieve quotes" (Just opts) True
+  where
+    opts =
+      ApplicationCommandOptionsSubcommands $
+        ApplicationCommandOptionSubcommandOrGroupSubcommand
+          <$> [ addQuoteAppComm,
+                showQuoteAppComm,
+                randomQuoteAppComm,
+                authorQuoteAppComm,
+                editQuoteAppComm
+              ]
+    addQuoteAppComm =
+      ApplicationCommandOptionSubcommand
+        "add"
+        "add a new quote"
+        [ ApplicationCommandOptionValueString "quote" "what the actual quote is" True (Left False),
+          ApplicationCommandOptionValueString "author" "who authored this quote" True (Left False)
+        ]
+    showQuoteAppComm =
+      ApplicationCommandOptionSubcommand
+        "show"
+        "show a quote by number"
+        [ ApplicationCommandOptionValueInteger "id" "the quote's number" True (Left True) (Just 1) Nothing
+        ]
+    randomQuoteAppComm =
+      ApplicationCommandOptionSubcommand
+        "random"
+        "show a random quote"
+        []
+    authorQuoteAppComm =
+      ApplicationCommandOptionSubcommand
+        "author"
+        "show a random quote by an author"
+        [ApplicationCommandOptionValueString "author" "whose quotes do you want to see" True (Left False)]
+    editQuoteAppComm =
+      ApplicationCommandOptionSubcommand
+        "edit"
+        "edit a quote"
+        [ ApplicationCommandOptionValueInteger "quoteid" "the id of the quote to edit" True (Left False) Nothing Nothing,
+          ApplicationCommandOptionValueString "quote" "what the actual quote is" False (Left False),
+          ApplicationCommandOptionValueString "author" "who authored this quote" False (Left False)
+        ]
+
+quoteApplicationCommandRecv :: Interaction -> DatabaseDiscord ()
+quoteApplicationCommandRecv i@InteractionApplicationCommand {interactionDataApplicationCommand = InteractionDataApplicationCommandChatInput {interactionDataApplicationCommandOptions = Just (InteractionDataApplicationCommandOptionsSubcommands [InteractionDataApplicationCommandOptionSubcommandOrGroupSubcommand subc])}} = case subcname of
```

Comment:
why is this line so long

Reply:
karl doesn't like short names and benji has word wrap on by default (I'll fix this)